### PR TITLE
Add tool to keep CLI usage in `index.md` in sync

### PR DESCRIPTION
**Pull request by florimondmanca**
This PR adds a simple Python-based tool to dump the output of `uvicorn --help` into the docs home page. Also supports a `--check` mode for CI purposes.

**Motivation**
While working on #875 I realized the CLI usage there was out of sync with the current options, and I thought adding an automated validation tool was the best way to keep it in sync moving forward. :-)

I could expand this later on to also validate that all options are listed in the "Settings" page, but that's a different thing so I'll keep it for another PR.

### CODE REVIEW

File: tools/gen_cli_usage.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        content = f.read()
+
+    lines = content.splitlines()
+    marker_lineno = lines.index("<!-- gen_cli_usage:marker -->")
+    assert lines[marker_lineno + 1] == "```"
+    start = marker_lineno + 2
+    end = next(
+        lineno for lineno, line in enumerate(lines[start:], start) if line == "```"
+    )
+    lines = lines[:start] + get_usage_lines() + lines[end:]
```
Comment (florimondmanca) [inline tools/gen_cli_usage.py]: Honestly, these kinds of < 10 LOC scripts that do useful tasks yet remain pretty readable is why I love Python so much. ❤️

**Comment by euri10** [comment]
lgtm, 
but why you dont use mkdocs-click you wrote ???

**Review by euri10 — APPROVED** [review]

**Comment by florimondmanca** [comment]
> mkdocs-click

😳 Absolutely! Definitley worth considering, I'll look at what that looks like. https://github.com/DataDog/mkdocs-click Though right now my thoughts would be "a 40-LOC script vs a new dependency?" tends to lean towards the former…

Also a note to myself that we also have a listing of the CLI options in the [Deployment docs](https://www.uvicorn.org/deployment/) (for some reason).

**Comment by florimondmanca** [comment]
@euri10 After considering it, I won't be using `mkdocs-click`. It adds an `h1` with the name of the Click command, which we don't need, and it's an extra dependency which means if anything changes there then we might need to tweak things here, too. A simple script inside the repo is much more resilient. :-)

I updated the script to be a bit more robust, feel free to take a new look!

**Comment by euri10** [comment]
nothing to add @florimondmanca  ! let's merge and see it deployed :)

**Pull request merged**